src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -1277,9 +1277,8 @@ MemorySegment reinterpret(long newSize,
      * @throws IllegalArgumentException if the size of the string is greater than the
      *         largest string supported by the platform
      * @throws IndexOutOfBoundsException if {@code offset < 0}
-     * @throws IndexOutOfBoundsException if {@code offset > byteSize() - (B + 1)}, where
-     *         {@code B} is the size, in bytes, of the string encoded using UTF-8 charset
-     *         {@code str.getBytes(StandardCharsets.UTF_8).length})
+     * @throws IndexOutOfBoundsException if no {@code '\0'} terminator is present in
```

Comment:
Maybe we should say "string terminator" instead of `\0`. Because the specified byte sequence of the string terminator is charset specific. So maybe:

> if no string terminator (e.g. `\0`) is present ... The size of the string terminator depends on the selected charset (... follow with text from the previous javadoc ... )

Reply:
(probably we should only expand on the charset digression in the `getString` method which takes a charset